pycheribuild/projects/cross/gkermit.py

```diff
@@ -0,0 +1,63 @@
+#
+# Copyright (c) 2021 Microsoft
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+# OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+# SUCH DAMAGE.
+
+from ..project import ExternallyManagedSourceRepository
+from .crosscompileproject import CrossCompileMakefileProject, DefaultInstallDir
+
+
+class BuildGKermit(CrossCompileMakefileProject):
+    build_in_source_dir = False
+    cross_install_dir = DefaultInstallDir.ROOTFS_OPTBASE
+    native_install_dir = DefaultInstallDir.BOOTSTRAP_TOOLS
+    repository = ExternallyManagedSourceRepository()
+
+    def clean(self):
+        return self.async_clean_directory(self.build_dir, keep_root=True)
+
+    def update(self):
+        filename = "gku201.tar.gz"
+        sha256 = "19f9ac00d7b230d0a841928a25676269363c2925afc23e62704cde516fc1abbd"
+        url_prefix = "https://www.kermitproject.org/ftp/kermit/archives/"
+
+        self.makedirs(self.source_dir)
+        if not (self.source_dir / filename).is_file():
+            self.download_file(self.source_dir / filename, url=url_prefix + filename, sha256=sha256)
+            self.run_cmd(["tar", "xzvf", self.source_dir / filename, "-C", self.source_dir])
+
+    def needs_configure(self):
+        return not (self.build_dir / "makefile").exists()
+
+    def configure(self, **kwargs):
+        if not self.should_run_configure():
+            return
+
+        self.write_file(self.build_dir / "makefile", overwrite=True, mode=0o644, contents=f"""
```

Comment:
What's wrong with upstream's Makefile?

Reply:
It expects in-source-tree builds and to be edited to set `CC`.

Reply:
`build_via_symlink_farm = True` deals with the former (it has its own issues, but those shouldn't matter for something like kermit that's not frequently changing), `make CC=foo` deals with the latter.